lib/logstash/plugin_mixins/http_client/obsolete_ssl_config_support.rb

```diff
@@ -0,0 +1,22 @@
+module LogStash::PluginMixins::HttpClient
+  module ObsoleteSslConfigSupport
+    def self.included(base)
+      fail ArgumentError unless base <= LogStash::PluginMixins::HttpClient::Implementation
+
+      require 'logstash/plugin_mixins/normalize_config_support'
+      base.include(LogStash::PluginMixins::NormalizeConfigSupport)
+
+      base.config :cacert, :obsolete => 'Use `ssl_certificate_authorities` instead'
+      base.config :client_cert, :obsolete => 'Use `ssl_certificate` instead'
+      base.config :client_key, :obsolete => 'Use `ssl_key` instead'
+      base.config :keystore, :obsolete => 'Use `ssl_keystore_path` instead'
+      base.config :keystore_type, :obsolete => 'Use `ssl_keystore_type` instead'
+      base.config :truststore, :obsolete => 'Use `ssl_truststore_path` instead'
+      base.config :truststore_type, :obsolete => 'Use `ssl_truststore_type` instead'
+
+      # Retain validation for password types to avoid inadvertent information disclosure
```

Comment:
An, this is interesting. I was surprised to find that marking a config option as `:obsolete` would still do input validation. Sounds like there is also some obfuscation as well when `:validate` is type `:password`.

Reply:
Yeah, it's a little "magical" and non-obvious - https://github.com/elastic/logstash/blob/a94659cf82d7e62c5a5f550df3e4e5bd0c6cf3e1/logstash-core/lib/logstash/config/mixin.rb#L608-L615